### PR TITLE
#254 The changes are necessary for the validation of Expand/Collapse …

### PR DESCRIPTION
**Pull request by mmilitaru**
…context menu.

### Prerequisites

- [ ] I have written a descriptive pull-request title
- [ ] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [ ] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [ ] I have provided test coverage for my change (where applicable)

### Description
#254 Fix for Expand/Collapse context menu.

### CODE REVIEW

File: BasicRdl/ViewModels/CategoryBrowserViewModel.cs [hunk 1]
```diff
@@ -152,6 +152,16 @@ public override void ComputePermission()
         public override void PopulateContextMenu()
         {
             base.PopulateContextMenu();
+
+            if (this.SelectedThing == null || this.SelectedThing.ContainedRows.Count == 0)
```
Comment (alexatrhea) [inline BasicRdl/ViewModels/CategoryBrowserViewModel.cs]: Can this not be abstracted to a higher base class? to avoid repeating this everywhere?

Comment (mmilitaru) [reply BasicRdl/ViewModels/CategoryBrowserViewModel.cs]: Agreed. Change applied

File: CDP4Composition/Mvvm/BrowserViewModelBase.cs [hunk 2]
```diff
@@ -373,7 +373,18 @@ public bool HasUpdateStarted
         public IRowViewModelBase<Thing> SelectedThing
         {
             get { return this.selectedThing; }
-            set { this.RaiseAndSetIfChanged(ref this.selectedThing, value); }
+            set
+            {
+                if (value == null || value.ContainedRows.Count == 0)
+                {
```
Comment (alexatrhea) [inline CDP4Composition/Mvvm/BrowserViewModelBase.cs]: this logic should not be in the setter here.

Comment (mmilitaru) [reply CDP4Composition/Mvvm/BrowserViewModelBase.cs]: Remove the logic and checking directly under context menu creation

File: ProductTree/ViewModels/ProductTreeViewModel.cs [hunk 3]
```diff
@@ -187,7 +187,7 @@ public bool IsDisplayShortNamesOn
             get { return this.isDisplayShortNamesOn; }
             private set { this.RaiseAndSetIfChanged(ref this.isDisplayShortNamesOn, value); }
         }
-        
+
```
Comment (alexatrhea) [inline ProductTree/ViewModels/ProductTreeViewModel.cs]: Remove consecutive new line

**Pull request merged**